tests/test_destination.py

```diff
@@ -0,0 +1,368 @@
+        print(f'num_excs: {us.num_excs}')
+        print(f'updated_cols: {us.updated_cols}')
+        print(f'cols_with_excs: {us.cols_with_excs}')
+        print(us.row_count_stats)
+        print(us.cascade_row_count_stats)
+        print('============================================================')
+
+    def test_dest_errors(self, reset_db: None) -> None:
+        t = pxt.create_table('test_dest_errors', schema={'img': pxt.Image})
+        valid_dest = 'tests/data/'
+
+        # Basic tests of the destination parameter: types and store / computed
+        with pytest.raises(excs.Error, match='must be a string or path'):
+            t.add_computed_column(img_rot=t.img.rotate(90), destination=27)
+
+        with pytest.raises(excs.Error, match='only applies to stored computed columns'):
+            t.add_computed_column(img_rot=t.img.rotate(90), stored=False, destination=valid_dest)
+
+        with pytest.raises(excs.Error, match='only applies to stored computed columns'):
+            _ = pxt.create_table('test_dest_bad', schema={'img': {'type': pxt.Image, 'destination': f'{valid_dest}'}})
+
+        # Test destination with a non-existent directory
+        with pytest.raises(excs.Error, match='does not exist'):
+            t.add_computed_column(img_rot=t.img.rotate(90), destination='non_existent_dir/img_rot')
+
+        # Test destination with a file path instead of a directory
+        with pytest.raises(excs.Error, match='must be a directory, not a file'):
+            t.add_computed_column(
+                img_rot=t.img.rotate(90), destination='tests/data/imagenette2-160/ILSVRC2012_val_00000557.JPEG'
+            )
+
+        # Test with invalid scheme
+        with pytest.raises(excs.Error, match='must be a valid URI to a supported'):
+            t.add_computed_column(img_rot=t.img.rotate(90), destination='https://anything/')
+
+    def parse_one(self, s: str, consider_object: bool) -> bool:
+        print(f'Parsing: {s}')
+        try:
+            r = MediaPath.parse_media_storage_addr(s, consider_object)
+            print(f'Success: {r!r}')
+            return True
+        except ValueError as e:
+            print(f'Error: {e!r}')
+            return False
+
+    def parse_two(self, s: str) -> bool:
+        r = self.parse_one(s, False)
+        r &= self.parse_one(s, True)
+        return r
+
+    def test_dest_parser(self, reset_db: None) -> None:
+        a_name = 'acct-name'
+        o_name = 'obj-name'
+        p_name1 = 'path-name'
+        p_name2 = 'path-name/path2-name'
+        r = True
+        for s in [
+            's3://container',
+            f'wasb://container@{a_name}.blob.core.windows.net',
+            f'https://{a_name}.blob.core.windows.net/container',
+            f'https://{a_name}.r2.cloudflarestorage.com/container',
+            'https://raw.github.com',
+            'file://dir1/dir2/dir3',
+            'dir1/dir2/dir3',
+        ]:
+            r &= self.parse_two(s)
```

Comment:
IMO this is flawed assertion design and will make things harder to debug in the event something fails. Better would be to have a separate assertion for each check.

I'd advocate for something like
```
def parse_uri(self, uri: str, consider_object: bool) -> None:
    try:
        ...
    except ValueError as e:
        raise AssertionError('Failed to parse URI: {url}') from e

def parse_uri_twice(self, uri: str) -> None:
    self.parse_uri(uri, True)
    self.parse_uri(uri, False)
```

And then
```
assert self.parse_uri_twice(...)
```
each time it's used. Note that I'm also using method and parameter names that are more likely to be meaningful to someone reading the code